train.py

```diff
@@ -27,31 +25,18 @@
     model_name_to_weights_download_fns,
     model_name_to_weights_export_fns,
     model_name_to_tokenizer,
-    models_config
+    models_config,
 )
 from torchtitan.optimizer import build_lr_schedulers, build_optimizers
 from torchtitan.parallelisms import models_parallelize_fns, ParallelDims
 from torchtitan.profiling import maybe_enable_memory_snapshot, maybe_enable_profiling
-
-
-def get_train_context(enable_loss_parallel: bool, enable_compiled_autograd: bool):
-    @contextlib.contextmanager
-    def context():
-        with contextlib.ExitStack() as stack:
-            if enable_loss_parallel:
-                stack.enter_context(torch.distributed.tensor.parallel.loss_parallel())
-            if enable_compiled_autograd:
-                stack.enter_context(
-                    torch._dynamo.utils.maybe_enable_compiled_autograd(True)
-                )
-            yield
-
-    return context
+from torchtitan.val import validate
+from typing import Any
 
 
 # Enable debug tracing on failure: https://pytorch.org/docs/stable/elastic/errors.html
 @record
-def main(job_config: JobConfig):
```

Comment:
This eventually needs to be changed thoroughly via more robust typing, but currently LSP complains about all attribute accesses into job_config object, because the attributes are set dynamically in config_manager. I think it's better to keep it as any, since this interface is really simple and the warnings from LSP are misleading.